src/eve/pattern_matching.py

```diff
@@ -0,0 +1,162 @@
+    item. Objects are compared using equality operator or if the left-hand-side
+    is an `ObjectPattern` its type and attributes are compared to the
+    right-hand-side object. Only the attributes of the `ObjectPattern` are used
+    for comparison, disregarding potential additional attributes of the
+    right-hand-side.
+    """
+    if type(a) is not type(b):
+        yield (path, f"Expected a value of type {type(a).__name__}, but got {type(b).__name__}")
+    elif a != b:
+        yield (path, f"Values are not equal. `{a}` != `{b}`")
+
+
+@get_differences.register
+def _(a: ObjectPattern, b: Any, path: str = "") -> Iterator[tuple[str, str]]:
+    if not isinstance(b, a.cls):
+        yield (
+            path,
+            f"Expected an instance of class {a.cls.__name__}, but got {type(b).__name__}",
+        )
+    else:
+        for k in a.fields.keys():
+            if not hasattr(b, k):
+                yield (path, f"Value has no attribute {k}.")
+            else:
+                yield from get_differences(a.fields[k], getattr(b, k), path=f"{path}.{k}")
+
+
+@get_differences.register
+def _(a: list, b: Any, path: str = "") -> Iterator[tuple[str, str]]:
+    if not isinstance(b, list):
+        yield (path, f"Expected list, but got {type(b).__name__}")
+    elif len(a) != len(b):
+        yield (path, f"Expected list of length {len(a)}, but got length {len(b)}")
+    else:
+        for i, (el_a, el_b) in enumerate(zip(a, b)):
+            yield from get_differences(el_a, el_b, path=f"{path}[{i}]")
+
+
+@get_differences.register
+def _(a: dict, b: Any, path: str = "") -> Iterator[tuple[str, str]]:
+    if not isinstance(b, dict):
+        yield (path, f"Expected dict, but got {type(b).__name__}")
+    elif set(a.keys()) != set(b.keys()):
+        a_min_b = set(a.keys()).difference(b.keys())
+        b_min_a = set(b.keys()).difference(a.keys())
+        if a_min_b:
+            missing_keys_str = "`" + "`, `".join(map(str, a_min_b)) + "`"
+            yield (
+                path,
+                f"Expected dictionary with keys `{'`, `'.join(map(str, a.keys()))}`, but the following keys are missing: {missing_keys_str}",
+            )
+        if b_min_a:
+            extra_keys_str = "`" + "`, `".join(map(str, b_min_a)) + "`"
+            yield (
+                path,
+                f"Expected dictionary with keys `{'`, `'.join(map(str, a.keys()))}`, but the following keys are extra: {extra_keys_str}",
+            )
+    else:
+        for k, v_a, v_b in zip(a.keys(), a.values(), b.values()):
+            yield from get_differences(v_a, v_b, path=f'{path}["{k}"]')
+
+
+# TODO(tehrengruber): The ModuleWrapper has some magicness with it that could be confusing to the user. Do we really
+#  want this? Should we maybe use something more in the line of `P[foo_ir.Foo](...)`
+@dataclass(frozen=True)
+class ModuleWrapper:
+    """
+    Small wrapper to conveniently create `ObjectPattern`s for classes of a module.
```

Comment:
```suggestion
    """Small wrapper to conveniently create `ObjectPattern`s for classes of a module.
```

One single convention is still better than mixing, even if it is the ugliest convention.